modules/apps/frontend-css/frontend-css-variables-web/src/test/resources/com/liferay/frontend/css/variables/web/internal/theme/css-variables.json

```diff
@@ -0,0 +1,14 @@
+{
+	"variables": {
+		"background": {
+			"label": "Background Color",
```

Comment:
So... themes can't have ResourceBundles, do they?

I'm wondering if we could make this go through `LanguageUtil.get()` and avoid having to support localization in a json... this feels like a weird thing to build in here, though granted, we've done this in the past...

Thinking our efforts might be more valuable in figuring out how to provide a `ResourceBundle` for this than adding multilang support in here...

Reply:
No, they can't because they are WAR files. I will investigate how easy it is to instantiate a ResourceBundle on the fly or if there's something available...

Reply:
There's no way to add language keys to WAR files. So we may:

1. Leave localization in this JSON
2. Move the localization to properties files and create some sort of service that reads them from WAR files and instantiates `ResourceBundle`s
3. Migrate themes to JAR files
4. Don't localize

4 doesn't seem an option.

3 we already investigated it and was way more difficult than we thought. It would also imply migrating the toolkit.

2 is a good thing if we know for sure that we won't do 3 in the near future. The idea is that WAR files are kind of deprecated, so there's no point in making the migration more difficult by supporting newer things. However, WAR files will always work, and themes don't need Java classes, so I don't see it so bad to leave themes as WARs forever.

What do you think?

Reply:
BTW, for 2 we would use the same techniques we use in the JS Portlet Extender (instantiate an OSGi service on the fly configured by a certain file in the bundler/WAR).